pkg/cluster/deploy.go

```diff
@@ -30,3 +49,99 @@ func (u *simpleUpgrader) Deploy(ctx context.Context, cs *api.OpenShiftManagedClu
 	}
 	return nil
 }
+
+func (u *simpleUpgrader) initializeUpdateBlob(ctx context.Context, cs *api.OpenShiftManagedCluster, azuretemplate map[string]interface{}) error {
```

Comment:
I think you're going to find that you need to split this function into two parts.  The hashing part (function 1) needs to be called on create and update; the initialization of the blob (function 2) only on create.  However there is also similar code used for initialization of newly created vms on update.  I wonder whether function 2 and that code can be combined.